### PR TITLE
Support having multiple Prow job config files

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Support having multiple Prow job config files.

This should be merged after https://github.com/istio/test-infra/pull/2599 is merged.

/cc @chaodaiG 
/hold

### CODE REVIEW

**Comment by chizhg** [comment]
I have tested it in staging Prow and it works.

File: config/prow_common.mk [hunk 1]
```diff
@@ -56,42 +58,17 @@ get-cluster-credentials:
...
-	diff "${OLD_YAML_CONFIG}" "${NEW_YAML_CONFIG}" --color=auto || true
-	@echo "Inspect uploaded config file at: ${NEW_YAML_CONFIG}"
-	@echo "Old config file saved at: ${GCS_DEST}"
-endif
-
-# Update prow plugins
-update-prow-plugins: confirm-master
-	$(SET_CONTEXT)
-	kubectl create configmap plugins --from-file=plugins.yaml=$(PROW_PLUGINS) --dry-run --save-config -o yaml | kubectl apply -f -
+	python3 <(curl -sSfL https://raw.githubusercontent.com/istio/test-infra/master/prow/recreate_prow_configmaps.py) \
```
Comment (chaodaiG) [inline config/prow_common.mk:66]: This is from istio/test-infra master branch, Is this stable enough?

Comment (chizhg) [reply config/prow_common.mk:66]: It should be, but I can change it to use the specific commit like `https://github.com/istio/test-infra/blob/48d88083edbccd5e7ca037dd9b9f2b6d19f5359a/prow/recreate_prow_configmaps.py` if you think it's more preferable, but I guess we will need to change it again soon as this script can be moved to `oss-test-infra` to be shared.

Comment (chaodaiG) [reply config/prow_common.mk:66]: I guess it's ok to stick with master branch, depends on how active it's been updated

**Comment by chaodaiG** [comment]
/lgtm
/hold

feel free unhold

**Comment by chizhg** [comment]
The PR in istio test-infra has been merged.
/unhold

**Pull request merged**